kong/templates/nginx_kong.lua

```diff
@@ -93,7 +93,9 @@ server {
 > if proxy_ssl_enabled then
     ssl_certificate ${{SSL_CERT}};
     ssl_certificate_key ${{SSL_CERT_KEY}};
-    ssl_protocols TLSv1.1 TLSv1.2 TLSv1.3;
+> if ssl_protocols then
+    ssl_protocols ${{SSL_PROTOCOLS}};
```

Comment:
In several PRs already, we have discussed the topic of adding new Kong configuration properties mapping to Nginx directives. It has been established each time that this model does not work well for us, nor for our users.

If I recall correctly, we had agreed at least twice already that:

1. If new Nginx directives were to be added to our template, we should leverage *dynamically injected nginx directives*: add a new "Nginx directives" section to the `kong.conf.default` template, and add a new default property of, e.g. `nginx_http_large_client_header_buffers = 8 24k`. This not only allows us to specify our own defaults for Nginx directives, but it also advertises dynamically injected Nginx directives, *and* allows us to move away from the current paradigm of adding a configuration property for each directive, which is painful for several reasons.
2. That we should deprecate our existing 1-1 Kong configuration properties in favor of new default `nginx_*` properties resulting in underlying dynamic injections out of the box.

Can we invest some time in the above plan instead of digging ourselves a slightly deeper hole? :sweat_smile: